specificatie/openapi.yaml

```diff
@@ -572,11 +374,106 @@ components:
       required: true
       schema:
         type: "string"
-        pattern: "^[0-9]*$"
-        maxLength: 9
-        minLength: 9
+        pattern: "^[0-9]{9}$"
         example: "555555021"
   schemas:
+    IngeschrevenPersoonQuery:
+      type: "object"
+#      required:
+#      - fields
+      properties:
```

Comment:
Nu ik dit zie, zou ik ervoor kiezen om de fields veld hier op te nemen en niet als query parameter. Dit zou ik ook doen voor de paging parameters, die ik niet ben tegengekomen

Reply:
Hebben we afgesproken paging op te gaan nemen dan ? 
Dat is mij dan ontschoten. Paging zit nu ook niet in BRP-bevragen. 

Als we de fields opnemen in de requestbody wijken we af van hoe dat in de BAG gedaan wordt. We kunnen ook bij de BAG voorstellen alle parameters op te nemen in de Requestbody.  Het lijkt me wel van belang dat binnen de Haal Centraal API's dit op een consistente wijze gedaan wordt. 
De landelijke API-strategie zegt hier niets over.

Reply:
> 
> 
> Hebben we afgesproken paging op te gaan nemen dan ? Dat is mij dan ontschoten. Paging zit nu ook niet in BRP-bevragen.
> 
>
Paging is voorgesteld om de payload van een response te beperken. Dit om te voorkomen dat er onbedoeld het netwerk wordt beperkt. Als er een reden is om dit niet te doen, dan hoor ik het graag, zodat ik het kan toevoegen aan de threat model.

> Als we de fields opnemen in de requestbody wijken we af van hoe dat in de BAG gedaan wordt. We kunnen ook bij de BAG voorstellen alle parameters op te nemen in de Requestbody. Het lijkt me wel van belang dat binnen de Haal Centraal API's dit op een consistente wijze gedaan wordt. De landelijke API-strategie zegt hier niets over.

Bij een post zou ik ervoor kiezen om alle query parameters in de body te stoppen. Anders moet ik in code elke keer gaan uitzoeken wat in de request body moet en wat als query parameter. Wat is de design decision bij BAG om niet alle query parameters in de request body te stoppen?

Reply:
> Alle andere parameters staan als query-parameters bij de Post gedefinieerd.

Volgens mij is dit niet waar. Er zijn wel andere properties in de filter objecten. De enige reden die ik kan opmaken is dat query parameters als fields en expand niet worden gezien als filters die zijn gebaseerd op resource properties.
Echter, naar mijn mening heeft het weinig zin om deze parameters in de url op te nemen, want je kan niet door linken/klikken zonder de request body. Je ziet dan wel de fields, expand en page waarde in de url, maar je weet niet wat de andere gebruikte filters zijn.

Reply:
Ik kijk naar pull request 470: https://github.com/VNG-Realisatie/Haal-Centraal-BAG-bevragen/pull/470. De nieuwe specificatie

Reply:
> > Hebben we afgesproken paging op te gaan nemen dan ? Dat is mij dan ontschoten. Paging zit nu ook niet in BRP-bevragen.
> 
> Paging is voorgesteld om de payload van een response te beperken. Dit om te voorkomen dat er onbedoeld het netwerk wordt beperkt. Als er een reden is om dit niet te doen, dan hoor ik het graag, zodat ik het kan toevoegen aan de threat model.
> 
> > Als we de fields opnemen in de requestbody wijken we af van hoe dat in de BAG gedaan wordt. We kunnen ook bij de BAG voorstellen alle parameters op te nemen in de Requestbody. Het lijkt me wel van belang dat binnen de Haal Centraal API's dit op een consistente wijze gedaan wordt. De landelijke API-strategie zegt hier niets over.
> 
> Bij een post zou ik ervoor kiezen om alle query parameters in de body te stoppen. Anders moet ik in code elke keer gaan uitzoeken wat in de request body moet en wat als query parameter. Wat is de design decision bij BAG om niet alle query parameters in de request body te stoppen?

Pagineren is ook genoemd ten behoeve van dataminimalisatie. Als de persoon die je nodig hebt al in de eerste 10 zoekresultaten zit, is het niet nodig ook resultaten 11 e.v. te krijgen. Dus moet er bij zoeken elke keer een relatief kleine set resultaten worden teruggegeven. Ik zou dan dus denken aan default pageSize 10.

Ik ben het eens met @MelvLee dat het duidelijker is alle queryparameters naar de request body te verhuizen.

Reply:
> Gaan we in de pilot ook Historie, bewoning en reisdocumenten bouwen ? Want dat betekent dat voor al die API's dan ook POST varianten moeten worden gemaakt

Het antwoord daarop is Ja

> Tevens realiseer ik me nu dat bij het toepassen van de POST variant in deze situatie het faciliteren van de discoverability (samen met HAL) komt te vervallen.

Dat is wat @MelvLee ook aangeeft. In de POST kan de _links verwijderd worden, zowel op hoofdniveau als de ingeschrevenpersonen in de resultaatcollectie. De collectie kan dan zonder de _embedded worden geleverd.

Reply:
Voorstel Paging toegevoegd in de component IngeschrevenPersoonCollectie.